src/cmd/linuxkit/moby/build/docker.go

```diff
@@ -24,15 +24,17 @@ func dockerRun(input io.Reader, output io.Writer, img string, runEnv []string, i
 	}
 
 	env := os.Environ()
-
-	// Pull first to avoid https://github.com/docker/cli/issues/631
-	pull := exec.Command(docker, "pull", img)
-	pull.Env = env
-	if err := pull.Run(); err != nil {
-		if exitError, ok := err.(*exec.ExitError); ok {
-			return fmt.Errorf("docker pull %s failed: %v output:\n%s", img, err, exitError.Stderr)
+	// Only pull if its not local.
```

Comment:
I don't think there is any need for this change. When you do `docker pull`, it pulls the tag, checks the manifest, and if its digest matches the local, it doesn't pull anything. This is effectively a no-cost op.

Reply:
Ah, that is a good point. It works if the image already is on the registry. `docker pull` explicitly means, "check the registry and get the latest manifest hash." If it is not published, it will not work.

I wouldn't change that for the use case, or at least not as part of an unrelated PR. You can test it manually via `docker build` like you clearly did (or more likely `linuxkit pkg build`), and then do the 2-step I showed in the other comment:

1. `linuxkit build --format tar-kernel-initrd`
2. `cat test-initrd.tar | docker run -i linuxkit/mkimage-raw-efi:<your_tag> > test-efi.img`

Reply:
wiki not so much, but the `docs/` directory is there, no one will complain about contributions. 😄